object.go

```diff
@@ -375,11 +375,15 @@ func (d *Decoder) decInstance(typ reflect.Type, cls classInfo) (interface{}, err
 	for i := 0; i < len(cls.fieldNameList); i++ {
 		fieldName := cls.fieldNameList[i]
 
-		index, fieldStruct, err := findField(fieldName, typ)
+		index, fieldStruct, err := findFieldWithCache(fieldName, typ)
 		if err != nil {
 			return nil, perrors.Errorf("can not find field %s", fieldName)
 		}
 
+		if fieldStruct == nil {
```

Comment:
the fieldStruct won't be nil if err is nil, this check is redundant.  And should it return error when fieldStruct is nil?

Reply:
because of #183, I avoid to depend on
 err.